tests/testsuite_default_SharedObject.py

```diff
@@ -0,0 +1,157 @@
+
+
+    # SharedMap Tests
+    # Full Initialization
+    #   single - only ID 0 initializes array
+    #   multi - also has ID N-1 initialize array
+    #     no conflict - both IDs write same data
+    #     conflict - IDs write different data
+    # Partial initialize (everyone initializes their own portion)
+    #   late_write = true
+    #   do not publish = true
+    #   nopub and late_write true
+    #   double_initialize
+    def test_SharedObject_map_full_single(self):
+        self.sharedobject_test_template("map_full_single", 0, "--param=object_type:map --param=num_entities:12 --param=full_initialization:true")
+
+    def test_SharedObject_map_full_multi(self):
+        self.sharedobject_test_template("map_full_multi", 0, "--param=object_type:map --param=num_entities:12 --param=full_initialization:true --param=multiple_initializers:true")
+
+    def test_SharedObject_map_full_multi_conflict(self):
+        self.sharedobject_test_template("map_full_multi_conflict", 1, "--param=object_type:map --param=num_entities:12 --param=full_initialization:true --param=multiple_initializers:true --param=conflicting_write:true")
+
+    def test_SharedObject_map_partial(self):
+        self.sharedobject_test_template("map_partial", 0, "--param=object_type:map --param=num_entities:12 --param=full_initialization:false")
+
+    def test_SharedObject_map_partial_nopub(self):
+        self.sharedobject_test_template("map_partial_nopub", 0, "--param=object_type:map --param=num_entities:12 --param=full_initialization:false --param=publish:false")
+
+    def test_SharedObject_map_partial_late(self):
+        self.sharedobject_test_template("map_partial_late", 1, "--param=object_type:map --param=num_entities:12 --param=full_initialization:false --param=late_write:true")
+
+    def test_SharedObject_map_partial_late_nopub(self):
+        self.sharedobject_test_template("map_partial_late_nopub", 1, "--param=object_type:map --param=num_entities:12 --param=full_initialization:false --param=late_write:true --param=publish:false")
+        
+    def test_SharedObject_map_partial_doubleinit(self):
+        self.sharedobject_test_template("map_partial_doubleinit", 1, "--param=object_type:map --param=num_entities:12 --param=full_initialization:false --param=double_initialize:true")
+        
+#####
+
+    def sharedobject_test_template(self, testtype, exp_rc, options):
+        testsuitedir = self.get_testsuite_dir()
+        outdir = test_output_get_run_dir()
+
+        model_options = '--model-options="{0}"'.format(options)
+        print(model_options)
+
+        # Set the various file paths
+        sdlfile = "{0}/test_SharedObject.py".format(testsuitedir)
+        #reffile = "{0}/sharedobject_tests/refFiles/test_{1}.out".format(testsuitedir, testtype)
+        outfile = "{0}/test_SharedObject_{1}.out".format(outdir, testtype)
```

Comment:
The test run is displaying output to the console.  We need to add the err file to the run_sst
```
errfile = "{0}/test_SharedObject_{1}.err".format(outdir, testtype)

self.run_sst(sdlfile, outfile, errfile, other_args=model_options, expected_rc = exp_rc)
```